packages/terra-functional-testing/src/services/wdio-visual-regression-service/methods/LocalCompare.js

```diff
@@ -1,43 +1,132 @@
 import fs from 'fs-extra';
 import resemble from 'node-resemble-js';
 import _ from 'lodash';
+import path from 'path';
 import logger from '@wdio/logger';
 
 import BaseCompare from './BaseCompare';
+import terraViewports from '../../../utils/viewports';
 
 const log = logger('wdio-visual-regression-service:LocalCompare');
 
+const TEST_ID_REGEX = /\[([^)]+)\]/;
+
 export default class LocalCompare extends BaseCompare {
   constructor(options = {}) {
     super();
-    this.getScreenshotFile = options.screenshotName;
-    this.getReferencefile = options.referenceName;
-    this.getDiffFile = options.diffName;
-    this.misMatchTolerance = _.get(options, 'misMatchTolerance', 0.01);
-    this.ignoreComparison = _.get(options, 'ignoreComparison', 'nothing');
+    const {
+      baseScreenshotDir,
+      formFactor,
+      ignoreComparison,
+      locale,
+      misMatchTolerance,
+      theme,
+    } = options;
+
+    this.baseScreenshotDir = baseScreenshotDir;
+    this.formFactor = formFactor;
+    this.locale = locale;
+    this.theme = theme;
+    this.misMatchTolerance = misMatchTolerance;
+    this.ignoreComparison = ignoreComparison;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  createTestName(fullName) {
+    const matches = TEST_ID_REGEX.exec(fullName);
+
+    // If test ID is provided, use the ID for a shorter test name, otherwise use the full name
+    let name = matches ? matches[1] : fullName.trim();
+
+    // Remove white space
+    name = name.replace(/[\s+.]/g, '_');
+
+    // Remove windows reserved characters. See: https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#naming_conventions
+    // eslint-disable-next-line no-useless-escape
+    name = name.replace(/[\?\<\>\/\\|\*\"\:\+\.]/g, '-');
+
+    return name;
+  }
+
+  getScreenshotName(context) {
+    const { name } = (context.options || {});
+    const parentName = this.createTestName(context.test.parent);
+    const testName = this.createTestName(name || context.test.title);
+
+    return `${parentName}[${testName}].png`;
+  }
+
+  getFormFactor(browserWidth) {
```

Comment:
Similar to: https://github.com/cerner/terra-toolkit-boneyard/blob/main/config/wdio/visualRegressionConf.js#L37